### ISSUE TITLE
✅ 🐞 Bug Report: Template responsive controls bug

### ISSUE THREAD

**Issue by wayheming** [elementor/elementor#16467]
Hello,
When you use responsive controls inside the template and then show it somewhere on the page the responsive CSS is not generated in the template's CSS file.
BUT when you first open this template single page (preview) the CSS is generated correctly. It doesn't work well only when the template is a part of the page.
Steps to reproduce:
1. Create a template, add some elements and set some padding values for mobile devices https://monosnap.com/file/PH8t9JDeXLYTkh0z1sB7mCjImtXi7v
2. Add this template's shortcode on any page https://monosnap.com/file/wycbO3UaklPBlUJjRLMSIjnWaLM2I0
3. Change the padding value in the template again.  And save the template. https://take.ms/9ZfH1D
4. Reload the page with the template and see that responsive settings are not applied https://take.ms/zRUYW
Here is a video that demonstrates the problem. https://monosnap.com/file/VSHSz4HmP4QB9SF43lDsSd89t8PEQr

<details>
<summary>System Info</summary>
```

== Server Environment ==
	Operating System: Linux
	Software: TasteWP Official/2.1.4
	MySQL version: (Ubuntu) v8.0.26
	PHP Version: 7.4.23
	PHP Max Input Vars: 1000
	PHP Max Post Size: 60M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.8.1
	Site URL: https://shallowskirt.tastewp.com
	Home URL: https://shallowskirt.tastewp.com
	WP Multisite: No
	Max Upload Size: 60 MB
	Memory limit: 96M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Active

== Theme ==
	Name: Hello Elementor
	Version: 2.4.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.4.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.4.1
		Author: Elementor.com


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	


== Elements Usage ==
	
	section : 1
		text-editor : 1
	wp-page : 2
		text-editor : 2


== Log ==
	
Log: showing 5 of 52021-10-04 11:15:24 [info] elementor::elementor_updater Started 
2021-10-04 11:15:24 [info] Elementor/Upgrades - _on_each_version Start  
2021-10-04 11:15:24 [info] Elementor/Upgrades - _on_each_version Finished 
2021-10-04 11:15:25 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.4.3',
  'to' => '3.4.4',
)]
2021-10-04 11:15:25 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.4.3',
  'to' => '3.4.4',
)]

PHP: showing 1 of 1PHP: 2021-10-04 11:15:42 [notice X 1][../wp-content/plugins/elementor-pro/license/updater.php::83] Undefined index: new_version [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

```
</details>

**Comment by aureliePrd** [comment]
Hi there, 
I did not see this issue here, sorry. but no one answered @wayheming 
The bug is still here. I created a new issue #18379

**Comment by MichaelDarko** [comment]
Same here.

**Comment by gregolias** [comment]
As a workaround, you need to click on the “Preview” button while editing your footer HTML block. And always keep the preview window opened whiled customizing and saving your HTML blocks with Elementor.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core 3.15.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
